### PR TITLE
Add support for subpartition data export in MySQL and Oracle

### LINKED ISSUE #607: [MYSQL] Export data hangs for sub-partitions
When we have a sub partitioned table in mysql, the migration hangs in the `export-data` phase. Earlier it used to error out from the ora2pg side but now it goes through if we directly use it. Still hanging via voyager. Attaching a sample schema for reference.

```
CREATE TABLE subpartitioning_test (BILL_NO INT, sale_date DATE, cust_code VARCHAR(15), 
AMOUNT DECIMAL(8,2))
PARTITION BY List(MONTH(sale_date) )
SUBPARTITION BY HASH(TO_DAYS(sale_date))
SUBPARTITIONS 2 (
PARTITION p0 VALUES IN (1, 2),   
PARTITION p1 VALUES IN (3, 4),   
PARTITION p2 VALUES IN (5, 6),   
PARTITION p3 VALUES IN (7,8,9),
PARTITION p4 VALUES IN (10,11,12)
);   


INSERT INTO subpartitioning_test VALUES (1, '2013-01-02', 'C001', 125.56), 
(2, '2013-01-25', 'C003', 456.50), 
(3, '2014-02-15', 'C012', 365.00), 
(4, '2013-03-26', 'C345', 785.00), 
(5, '2013-04-19', 'C234', 656.00), 
(6, '2013-05-31', 'C743', 854.00), 
(11, '2013-05-31', 'C743', 854.00),
(7, '2013-06-11', 'C234', 542.00), 
(8, '2013-07-24', 'C003', 300.00), 
(9, '2013-08-02', 'C456', 475.20),
(10, '2013-11-02', 'C456', 475.20);


SELECT PARTITION_NAME, TABLE_ROWS FROM INFORMATION_SCHEMA.PARTITIONS WHERE TABLE_NAME='subpartitioning_test';


```

Also I cross-checked the data files created by ora2pg and voyager and they match with just `postdata.sql` extra on our side.

Last couple of lines in the log file:

```
2022-11-21 08:29:18 INFO mysql.go:38 Querying row count of table automation.subpartitioning_test
2022-11-21 08:29:18 INFO mysql.go:43 Table "automation.subpartitioning_test" has 11 rows.
2022-11-21 08:29:18 INFO exportDataStatus.go:191 Replacing actualRowCount=11 inplace of expectedRowCount=11 for table=automation.subpartitioning_test
```

### PR DESCRIPTION
**Pull request by rahulb-yb**
[Fixes #607]
- Adds support for exporting data for sub partitions in MySQL and Oracle. A new flag `--flatten-partitions` has been added to enable the same, which disables the partition-specific export of data. If a subpartition is detected while not using this flag, we error out and suggest that the user use this flag (or use exclude-table-list).
- As of now, this patch also contains the framework to support sub partitions without sacrificing partition-specific exported row counts for validation purposes. This framework, however, is disabled as it is locked behind the usage of the newly introduced flag above.
- Updated ora2pg commit hash to enable us to export data for sub partitions.
- Automation framework yet to be added

### CODE REVIEW

File: yb-voyager/cmd/exportDataStatus.go [hunk 1]
```diff
@@ -134,6 +156,9 @@ func exportDataStatus(ctx context.Context, tablesProgressMetadata map[string]*ut
 			if quit {
 				break
 			}
+			// if !tablesProgressMetadata[key].IsSubPartition {
```
Comment (rahulb-yb) [inline yb-voyager/cmd/exportDataStatus.go:159]: Remove commented code

File: installer_scripts/install-yb-voyager [hunk 2]
```diff
@@ -27,7 +27,7 @@ LOG_FILE=/tmp/install-yb-voyager.log
 
 # commit hash in main branch after release 23.2
 # issues #489 #492 #247 in ora2pg were fixed after this change
-ORA2PG_VERSION="3e27abba5939d5796419a1d835d6d4d0b2d5e01d"
+ORA2PG_VERSION="f08d47a4e5c20ff7d348e535ed05ab571b4bb350"
```
Comment (priyanshi-yb) [inline installer_scripts/install-yb-voyager:31]: We might have to update it to this `7b8f018352fb24ce46e368358178e6b542381cfd` as  there are these three fixes which also need to picked up and this commit other cover all these three fixes.
1. https://github.com/yugabyte/yb-voyager/issues/574
2. https://github.com/yugabyte/yb-voyager/issues/511
3. https://github.com/yugabyte/yb-voyager/issues/573

CC: @shubham-yb

Comment (rahulb-yb) [reply installer_scripts/install-yb-voyager:31]: The commit you have mentioned is after the commit past which ora2pg breaks in our automation tests (`8b64f1a543c9e63d3eff80d43b83a74298bfb104`). I have taken the commit right before this point to maximise the issues fixed within ora2pg

Comment (priyanshi-yb) [reply installer_scripts/install-yb-voyager:31]: Oh okay, I didn't knew the commit for our fix is breaking some other things. Cool then we are good here.

**Pull request closed without merging**